synapse/events/__init__.py

```diff
@@ -41,8 +41,12 @@ def get_dict(self):
     def is_outlier(self):
         return getattr(self, "outlier", False)
 
-    def is_invite_from_remote(self):
-        return getattr(self, "invite_from_remote", False)
+    def is_new_remote_event(self):
+        """Whether this is a new remote event, like an invite or an invite
+        rejection. This is needed as those events are marked as outliers, but
+        they still need to be processed.
+        """
+        return getattr(self, "new_remote_event", False)
```

Comment:
this is a funny name for it. For a start, this will get persisted in the db, so we'll end up with "new" remote events which are months old; secondly, what is remote about it?

How about "out_of_band_membership", or something?

Reply:
Aside: it would be nice to rebuild this thing with `attr.s` or something at some point, rather than the somewhat opaque `getattr` calls. Probably one for another time though.